lib/cocoapods/external_sources/podspec_source.rb

```diff
@@ -9,13 +9,18 @@ class PodspecSource < AbstractExternalSource
       def fetch(sandbox)
         title = "Fetching podspec for `#{name}` #{description}"
         UI.titled_section(title,  :verbose_prefix => '-> ') do
-          is_json = podspec_uri.split('.').last == 'json'
-          require 'open-uri'
-          begin
-            open(podspec_uri) { |io| store_podspec(sandbox, io.read, is_json) }
-          rescue OpenURI::HTTPError => e
-            status = e.io.status.join(' ')
-            raise Informative, "Failed to fetch podspec for `#{name}` at `#{podspec_uri}`.\n Error: #{status}"
+          podspec_path = Pathname(podspec_uri)
+          if podspec_path.exist?
+            store_podspec(sandbox, podspec_path, podspec_path.extname == '.json')
+          else
+            is_json = podspec_uri.split('.').last == 'json'
```

Comment:
What's the purpose of using two different pieces of logic to determine if the file is a JSON podspec?

```ruby
podspec_path.extname == '.json'
podspec_uri.split('.').last == 'json'
```